src/server/debug/server/server.go

```diff
@@ -101,9 +102,74 @@ func NewDebugServer(env Env) debug.DebugServer {
 		logLevel:      log.LogLevel,
 		grpcLevel:     log.GRPCLevel,
 	}
+	ticker := time.NewTicker(3 * time.Second)
+	query := `SELECT current_timestamp - query_start as runtime,
+                         datname,
+                         usename,
+                         client_addr,
+                         backend_xid,
+                         query
+		  FROM pg_stat_activity WHERE state = 'idle' AND current_timestamp - query_start > interval '10 seconds' ORDER BY runtime DESC;`
+	ctx := pctx.Background("debug server")
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				rows, err := s.database.QueryContext(ctx, query)
+				if err != nil {
+					log.Error(ctx, "query idle txns", zap.Error(err))
+				}
+				var buf bytes.Buffer
+				if err := s.writeRowsToJSON(rows, &buf); err != nil {
+					log.Error(ctx, "write idle txns query results", zap.Error(err))
+				}
+				log.Info(ctx, "idle txns", zap.String("txns", buf.String()))
+			}
+		}
+	}()
 	return s
 }
 
+func (s *debugServer) logDBState() {
```

Comment:
This appears to be dead code.

Reply:
If this is _not_ dead code, similar comments as above apply. It should almost certainly take a `context.Context` rather than allocating one itself, and the infinite loop should check against it.

Note that there is no need here to worry about how to indicate completion to the caller, since that is done simply by returning.